src/commands/sqlite.rs

```diff
@@ -445,8 +487,11 @@ pub fn find_database_link(db: &Database, label: &str) -> Option<Link> {
     })
 }
 
-pub fn database_has_link(db: &Database, link: &ResourceLabel) -> bool {
-    db.links.iter().any(|l| l == link)
+pub fn database_has_link(database: &Database, label: &str, app: &Option<String>) -> bool {
+    database
+        .links
+        .iter()
+        .any(|l| l.label == label && &l.app_name == app)
```

Comment:
Instead of taking in an owned string we can deref here
```suggestion
pub fn database_has_link(database: &Database, label: &str, app: Option<&str>) -> bool {
    database
        .links
        .iter()
        .any(|l| l.label == label && l.app_name.as_deref() == app)
}
```

Reply:
Done, thanks!